sw/host/tests/chip/pwrmgr/src/sleep_all_wakeups.rs

```diff
@@ -0,0 +1,57 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::Result;
+use clap::Parser;
+use std::time::Duration;
+
+use opentitanlib::test_utils::init::InitializeTest;
+use opentitanlib::uart::console::UartConsole;
+
+#[derive(Debug, Parser)]
+struct Opts {
+    #[command(flatten)]
+    init: InitializeTest,
+
+    /// Console receive timeout.
+    #[arg(long, value_parser = humantime::parse_duration, default_value = "600s")]
+    timeout: Duration,
+}
+
+fn main() -> Result<()> {
+    let opts = Opts::parse();
+    opts.init.init_logging();
+    let transport = opts.init.init_target()?;
+
+    let uart = transport.uart("console")?;
+    uart.set_flow_control(true)?;
+    let ioc0_pin = transport.gpio_pin("Ioc0")?;
+    loop {
+        let gs = UartConsole::wait_for(&*uart, r"Done|Issue WFI to enter sleep ([0-9]+)", opts.timeout)?;
+        if gs[0] == "Done" {
+            return Ok(());
```

Comment:
FYI: We have specific strings we conventionally wait for.

https://github.com/lowRISC/opentitan/blob/f419160a3d5e356a23038ca88d2cbb9a143d82d5/rules/opentitan_test.bzl#L55-L57

Reply:
I removed the Done string, and I match PASS instead. What is a good example for handling all the other failure cases? It would be a bunch of regex alternatives. I suppose I'd have to pass a failure_rx arg to wait_for? Also, opentitanlib is injecting a PASS or FAIL string. Does it wrap the Fail in ExitStatus::ExitFailure? I am calling `wait_for(..)?` so I expected the ? to catch the not-okay cases...